torch/csrc/jit/codegen/cuda/lower_allocation.cpp

```diff
@@ -113,10 +112,12 @@ class AllocationInserter : public kir::MutableIrVisitor {
       if (info.buffer->fuserTv()->axis(axis_i)->isReduction()) {
         continue;
       }
-      auto ca_id =
-          gpu_lower->lowerValue(fuser_tv->getComputeAtAxis(axis_i).first)
+      auto concrete_id =
+          gpu_lower
+              ->lowerValue(gpu_lower->caLoopMap().getConcreteMappedID(
+                  fuser_tv->axis(axis_i)))
               ->as<kir::IterDomain>();
-      init_dims.push_back(ca_id);
+      init_dims.push_back(concrete_id);
```

Comment:
@naoyam is this right? Should we allocate/init based on a non-local iteration domain size? What happens when compute at axis is non-broadcast but local id is?

Reply:
It should be safe to ignore broadcast IDs. I suspect that isn't simply taken into consideration in our current implementation.

This may be related: #530

Reply:
Hmm, broadcast *is* ignored in `createAllocExpr`. Maybe there is some bug around here.

Reply:
It seems that we do create an initialization expression with a broadcast loop but don't generate it.

https://github.com/csarofeen/pytorch/blob/20_12_3_devel/torch/csrc/jit/codegen/cuda/codegen.cpp#L593

So, it's no harm, but seems to make more sense to skip such domains at this point.

Reply:
I'll create a small PR to ignore broadcast domains. 

Aside from the case of local-broadcast axes, is there any other concern here?

Reply:
Please see #601

Reply:
Thank you.